### ISSUE TITLE
How to define my dataset label in Causalnex？Like DecisionTreeClassifier

### ISSUE THREAD

**Issue by ziyuwzf** [quantumblacklabs/causalnex#61]
When i use DecisionTreeClassifier,like below:

```
X_train, X_test, y_train, y_test = train_test_split(train, label, test_size=0.2, random_state=33) 
dtc = DecisionTreeClassifier(max_depth=12)
dtc.fit(X_train, y_train)
```

how can i define the label of dataset in the process of CausalNex train?

**Comment by ziyuwzf** [comment]
what's more,how can i define the label of dataset in the process of `from_pandas`?

**Comment by GabrielAzevedoFerreiraQB** [comment]
Thanks for the question and for the interest in CausalNex!

There are 2 "fit" phases in CausalNex, as opposed to other more traditional ML models (such as decision trees).

First, we learn the structure of the model: what the dependencies look like. We use `from_pandas` for that indeed, but, if we want our model to be truly causal, we normally have to consult domain experts to give their opinions too.

Second, we do the "Probability fitting". This fits the parameters of the model (much like Decision trees would). The method for this is `bn.fit_cpds`. There is no notion of "target variable" in the Bayesian Network formalism, so what this does is to fit parameters that model all the variables as a whole.

Bonus: to "predict" a node we do `bn.predict(data, node)` , where `node` is the variable you are trying to predict. This means that you can use your trained model to predict any variable in the model, including your target (`bn.predict(data,YOUR_TARGET_VAR_NAME)`).

This is in a nutshell, but it will become much clearer if you have time to follow our tutorial here:
https://causalnex.readthedocs.io/en/latest/03_tutorial/03_tutorial.html

**Comment by qbphilip** [comment]
Thanks @GabrielAzevedoFerreiraQB,

We currently have an experimental work stream on an sklearn wrapper that uses a target/"dependent variable" explicitly.
It will still use notears in the background but will have fit/predict API as well as feature importances.

**Comment by qbphilip** [comment]
Please have a look at the supervised sklearn wrappers `DAGClassifier` and `DAGRegressor` which will work similarly to the `DecisionTreeClassifier` you mention (also has `.score` and `.feature_importances_`).
You can use: `from causalnex.structure import DAGClassifier`. You will need the pytorch to run them.

The graph is stored in `model.graph_` for further inference and counterfactual steps.
